src/writer/_dirent.h

```diff
@@ -24,6 +24,12 @@
 
 #include "debug.h"
 
+// The DirentInfo tag is usefull to track what it the actual
+// value stored in the union. But it also means a extra byte.
+// It is usefull to have it when we are changing the dirent but
+// once everything is ok, it is useless to keep it.
+#define WITH_TAG 0
+
```

Comment:
With proper encapsulation of `DirentInfo` (as proposed in an earlier comment) this optimization should be transparent to client code of `DirentInfo`.